tests/thread/oethread_enc/enc.cpp

```diff
@@ -245,6 +245,20 @@ OE_ECALL void LockAndUnlockMutexes(void* arg)
     }
 }
 
+// Keep the enclave busy until we get TCS exhaustion
+OE_ECALL void TestTCSExhaustion(void* args_)
+{
+    TestTCSArgs* volatile args = (TestTCSArgs*)args_;
+    static oe_spinlock_t _tcs_lock = OE_SPINLOCK_INITIALIZER;
+
+    // Increment the number of threads only on getting the _tcs_lock
+    oe_spin_lock(&_tcs_lock);
+    args->num_threads++;
+    oe_spin_unlock(&_tcs_lock);
+    while (args->num_threads + args->num_out_threads < args->tcs_count)
```

Comment:
Can you describe the logic of this loop here?
It is not obvious what it is doing.

Reply:
Maybe rename the num_threads, num_out_threads?
Initially I thought num_threads was total number of threads.

Reply:
ok, will incorporate

Reply:
Changed the names of the variables to reflect the tcs bindings being used and requested. Also added comment to describe the logic of the loop. High level description is also in host.cpp before the routine.